### PR TITLE
Cleanup script enhancements

### PR DESCRIPTION
**Pull request by sparrc**
These changes are made to reflect the AWS image builder security best
practices documented here:
https://docs.aws.amazon.com/imagebuilder/latest/userguide/security-best-practices.html

There were a few files that were not deleted in the script above, so at
the end of the new cleanup script we will continue deleting those with
'rm' and will engage with AWS teams about updating the image builder
script to include those files.

Our script also does not allow configuration of each individual deletion
stage. This is not needed for the OSS ami recipes since users could just
customize this script to their liking if they are creating a custom AMI.

### Testing

tested using internal functional test framework

New tests cover the changes: no

### Description for the changelog

Enhancement - changed AMI build cleanup script to be in-line with [ec2 image builder best practices](https://docs.aws.amazon.com/imagebuilder/latest/userguide/security-best-practices.html)

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/cleanup.sh [hunk 1]
```diff
@@ -3,26 +3,177 @@ set -ex
...
+    sudo rm -rf /var/log/journal/*
+fi
+
+# delete a few items missed in https://docs.aws.amazon.com/imagebuilder/latest/userguide/security-best-practices.html
 sudo rm -rf \
-    /etc/hostname \
     /etc/machine-id \
-    /etc/resolv.conf \
-    /etc/ssh/ssh_host* \
     /home/ec2-user/.ssh/authorized_keys \
```
Comment (yinyic) [inline scripts/cleanup.sh:170]: This seems to have already been covered by the best practice script (starting line 65)

File: scripts/cleanup.sh [hunk 2]
```diff
@@ -3,26 +3,177 @@ set -ex
...
-    /root/.ssh/authorized_keys \
-    /var/lib/cloud/data \
-    /var/lib/cloud/instance \
-    /var/lib/cloud/instances \
-    /var/lib/cloud/sem \
-    /var/lib/dhclient/* \
+    /var/cache/yum \
+    /tmp/* \
     /var/lib/dhcp/dhclient.* \
```
Comment (yinyic) [inline scripts/cleanup.sh:173]: Curious how we know those files ought to be deleted and would not cause any side effects?

Comment (sparrc) [reply scripts/cleanup.sh:173]: most of this was originally just taken directly from the amazon-eks-ami cleanup process: https://github.com/awslabs/amazon-eks-ami/blob/master/scripts/install-worker.sh#L343-L361

File: scripts/cleanup.sh (hunk 1 above)
Comment (sparrc) [reply scripts/cleanup.sh:170]: true I missed that, thanks!

**Review by yinyic — APPROVED** [review]

**Review by lydiafilipe — APPROVED** [review]

**Pull request merged**